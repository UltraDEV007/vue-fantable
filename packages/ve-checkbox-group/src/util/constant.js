// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
