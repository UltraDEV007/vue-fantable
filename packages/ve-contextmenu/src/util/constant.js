// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
