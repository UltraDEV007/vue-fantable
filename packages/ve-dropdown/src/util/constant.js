// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
