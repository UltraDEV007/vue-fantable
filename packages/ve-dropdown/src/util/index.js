/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
