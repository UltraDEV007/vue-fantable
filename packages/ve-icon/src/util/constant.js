// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
