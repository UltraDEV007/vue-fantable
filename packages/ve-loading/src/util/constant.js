// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
