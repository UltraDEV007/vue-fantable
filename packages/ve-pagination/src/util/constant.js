// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
