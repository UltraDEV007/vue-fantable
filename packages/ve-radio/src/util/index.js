import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
