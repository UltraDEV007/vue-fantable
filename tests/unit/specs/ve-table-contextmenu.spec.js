import { mount } from '@vue/test-utils'
import veTable from '@P/ve-table/ve-table'
import { later } from '../util'
import { cloneDeep } from 'lodash-es'
import { vi, describe, it, expect, beforeEach, afterEach } from 'vitest'

describe('veTable contextmenu', () => {
  const TABLE_DATA = [
    {
      name: 'John',
      date: '1900-05-20',
      number: '32',
      address: 'No.1 Century Avenue, Shanghai',
      rowKey: 0,
    },
    {
      name: 'Dickerson',
      date: '1910-06-20',
      number: '676',
      address: 'No.1 Century Avenue, Beijing',
      rowKey: 1,
    },
    {
      name: 'Larsen',
      date: '2000-07-20',
      number: '76',
      address: 'No.1 Century Avenue, Chongqing',
      rowKey: 2,
    },
    {
      name: 'Geneva',
      date: '2010-08-20',
      number: '7797',
      address: 'No.1 Century Avenue, Xiamen',
      rowKey: 3,
    },
    {
      name: 'Jami',
      date: '2020-09-20',
      number: '8978',
      address: 'No.1 Century Avenue, Shenzhen',
      rowKey: 4,
    },
  ]

  const COLUMNS = [
    {
      field: '',
      key: 'a',
      title: '',
      width: 50,
      align: 'center',
      operationColumn: true,
      renderBodyCell: ({ row, column, rowIndex }, h) => {
        return ++rowIndex
      },
    },
    {
      field: 'name',
      key: 'name',
      title: 'Name',
      align: 'left',
      width: '15%',
    },
    {
      field: 'date',
      key: 'date',
      title: 'Date',
      align: 'left',
      width: '15%',
    },
    {
      field: 'number',
      key: 'number',
      title: 'Number',
      align: 'right',
      width: '30%',
    },
    {
      field: 'address',
      key: 'address',
      title: 'Address',
      align: 'left',
      width: '40%',
    },
  ]

  const HEADER_CONTEXTMENUS = [
    {
      type: 'CUT',
    },
    {
      type: 'COPY',
    },
    {
      type: 'SEPARATOR',
    },
    {
      type: 'EMPTY_COLUMN',
    },
    {
      type: 'SEPARATOR',
    },
    {
      type: 'LEFT_FIXED_COLUMN_TO',
    },
    {
      type: 'CANCEL_LEFT_FIXED_COLUMN_TO',
    },
    {
      type: 'RIGHT_FIXED_COLUMN_TO',
    },
    {
      type: 'CANCEL_RIGHT_FIXED_COLUMN_TO',
    },
  ]

  const BODY_CONTEXTMENUS = [
    {
      type: 'CUT',
    },
    {
      type: 'COPY',
    },
    {
      type: 'SEPARATOR',
    },
    {
      type: 'INSERT_ROW_ABOVE',
    },
    {
      type: 'INSERT_ROW_BELOW',
    },
    {
      type: 'SEPARATOR',
    },
    {
      type: 'REMOVE_ROW',
    },
    {
      type: 'EMPTY_ROW',
    },
    {
      type: 'EMPTY_CELL',
    },
  ]

  afterEach(() => {
    return new Promise((resolve) => {
      const contextmenuNodes = document.querySelectorAll(
        '.ve-contextmenu-node',
      );

      [].forEach.call(contextmenuNodes, (nodeEl) => {
        nodeEl.remove()
      })

      resolve()
    })
  })

  it('render', () => {
    const wrapper = mount(veTable, {
      props: {
        columns: COLUMNS,
        tableData: TABLE_DATA,
        rowKeyFieldName: 'rowKey',
        contextmenuHeaderOption: {
          beforeShow: ({
            isWholeRowSelection,
            selectionRangeKeys,
            selectionRangeIndexes,
          }) => {},
          afterMenuClick: ({
            type,
            selectionRangeKeys,
            selectionRangeIndexes,
          }) => {},
          contextmenus: HEADER_CONTEXTMENUS,
        },
        contextmenuBodyOption: {
          beforeShow: ({
            isWholeRowSelection,
            selectionRangeKeys,
            selectionRangeIndexes,
          }) => {},
          afterMenuClick: ({
            type,
            selectionRangeKeys,
            selectionRangeIndexes,
          }) => {},
          contextmenus: BODY_CONTEXTMENUS,
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('header contextmenu beforeShow callback', async () => {
    const mockBeforeShowFn = vi.fn()

    const tableData = cloneDeep(TABLE_DATA)

    const WRAPPER = mount(
      {
        render() {
          return (
            <div>
              <ve-table
                row-key-field-name="rowKey"
                columns={COLUMNS}
                table-data={tableData}
                contextmenu-header-option={
                  this.contextmenuHeaderOption
                }
              />
            </div>
          )
        },
        data() {
          return {
            contextmenuHeaderOption: {
              beforeShow: ({
                isWholeColSelection,
                selectionRangeKeys,
                selectionRangeIndexes,
              }) => {
                mockBeforeShowFn({
                  isWholeColSelection,
                  selectionRangeKeys,
                  selectionRangeIndexes,
                })
              },
              afterMenuClick: ({
                type,
                selectionRangeKeys,
                selectionRangeIndexes,
              }) => {
                //
              },
              contextmenus: HEADER_CONTEXTMENUS,
            },
          }
        },
      },
      // need attach to documnet
      { attachTo: document.body },
    )

    const firstTrThEl = WRAPPER.findAll('.ve-table-header-tr')[0]
      .findAll('.ve-table-header-th')[1]

    firstTrThEl.trigger('mousedown', {
      button: 3,
    })
    await later()
    firstTrThEl.trigger('contextmenu')
    await later()

    const contextmenuNodes = document.querySelectorAll(
      '.ve-contextmenu-node',
    )

    expect(contextmenuNodes.length).toBe(7)

    expect(mockBeforeShowFn).toHaveBeenCalledWith({
      isWholeColSelection: true,
      selectionRangeKeys: {
        endColKey: 'name',
        endRowKey: 4,
        startColKey: 'name',
        startRowKey: 0,
      },
      selectionRangeIndexes: {
        endColIndex: 1,
        endRowIndex: 4,
        startColIndex: 1,
        startRowIndex: 0,
      },
    })
  })

  it('header contextmenu afterMenuClick callback', async () => {
    const mockAfterMenuClickFn = vi.fn()

    const tableData = cloneDeep(TABLE_DATA)

    const WRAPPER = mount(
      {
        render() {
          return (
            <div>
              <ve-table
                row-key-field-name="rowKey"
                columns={COLUMNS}
                table-data={tableData}
                contextmenu-header-option={
                  this.contextmenuHeaderOption
                }
              />
            </div>
          )
        },
        data() {
          return {
            contextmenuHeaderOption: {
              beforeShow: ({
                isWholeColSelection,
                selectionRangeKeys,
                selectionRangeIndexes,
              }) => {
                //
              },
              afterMenuClick: ({
                type,
                selectionRangeKeys,
                selectionRangeIndexes,
              }) => {
                mockAfterMenuClickFn({
                  type,
                  selectionRangeKeys,
                  selectionRangeIndexes,
                })
              },
              contextmenus: HEADER_CONTEXTMENUS,
            },
          }
        },
      },
      // need attach to documnet
      { attachTo: document.body },
    )

    const firstTrThEl = WRAPPER.findAll('.ve-table-header-tr')[0]
      .findAll('.ve-table-header-th')[1]

    firstTrThEl.trigger('mousedown', {
      button: 3,
    })
    await later()
    firstTrThEl.trigger('contextmenu')
    await later()

    const contextmenuNodes = document.querySelectorAll(
      '.ve-contextmenu-node',
    )

    expect(contextmenuNodes.length).toBe(7)

    const contentmenuItemClickEvent = new MouseEvent('click', {
      view: window, // window
      bubbles: true,
      cancelable: true,
    })

    contextmenuNodes[2].dispatchEvent(contentmenuItemClickEvent)

    await later()

    expect(mockAfterMenuClickFn).toHaveBeenCalledWith({
      type: 'EMPTY_COLUMN',
      selectionRangeKeys: {
        endColKey: 'name',
        endRowKey: 4,
        startColKey: 'name',
        startRowKey: 0,
      },
      selectionRangeIndexes: {
        endColIndex: 1,
        endRowIndex: 4,
        startColIndex: 1,
        startRowIndex: 0,
      },
    })
  })

  it('body contextmenu beforeShow callback', async () => {
    const mockBeforeShowFn = vi.fn()

    const tableData = cloneDeep(TABLE_DATA)

    const WRAPPER = mount(
      {
        render() {
          return (
            <div>
              <ve-table
                row-key-field-name="rowKey"
                columns={COLUMNS}
                table-data={tableData}
                contextmenu-body-option={
                  this.contextmenuBodyOption
                }
              />
            </div>
          )
        },
        data() {
          return {
            contextmenuBodyOption: {
              beforeShow: ({
                isWholeRowSelection,
                selectionRangeKeys,
                selectionRangeIndexes,
              }) => {
                mockBeforeShowFn({
                  isWholeRowSelection,
                  selectionRangeKeys,
                  selectionRangeIndexes,
                })
              },
              afterMenuClick: ({
                type,
                selectionRangeKeys,
                selectionRangeIndexes,
              }) => {
                //
              },
              contextmenus: BODY_CONTEXTMENUS,
            },
          }
        },
      },
      // need attach to documnet
      { attachTo: document.body },
    )

    const firstTrTdEl = WRAPPER.findAll('.ve-table-body-tr')[2]
      .findAll('.ve-table-body-td')[2]

    firstTrTdEl.trigger('mousedown', {
      button: 3,
    })
    await later()
    firstTrTdEl.trigger('contextmenu')
    await later()

    expect(firstTrTdEl.classes()).toContain('ve-table-cell-selection')

    const contextmenuNodes = document.querySelectorAll(
      '.ve-contextmenu-node',
    )

    expect(contextmenuNodes.length).toBe(6)

    expect(mockBeforeShowFn).toHaveBeenCalledWith({
      isWholeRowSelection: false,
      selectionRangeKeys: {
        endColKey: 'date',
        endRowKey: 2,
        startColKey: 'date',
        startRowKey: 2,
      },
      selectionRangeIndexes: {
        endColIndex: 2,
        endRowIndex: 2,
        startColIndex: 2,
        startRowIndex: 2,
      },
    })
  })

  it('body contextmenu afterMenuClick callback', async () => {
    const mockAfterMenuClickFn = vi.fn()

    const tableData = cloneDeep(TABLE_DATA)

    const WRAPPER = mount(
      {
        render() {
          return (
            <div>
              <ve-table
                row-key-field-name="rowKey"
                columns={COLUMNS}
                table-data={tableData}
                contextmenu-body-option={
                  this.contextmenuBodyOption
                }
              />
            </div>
          )
        },
        data() {
          return {
            contextmenuBodyOption: {
              beforeShow: ({
                isWholeColSelection,
                selectionRangeKeys,
                selectionRangeIndexes,
              }) => {
                //
              },
              afterMenuClick: ({
                type,
                selectionRangeKeys,
                selectionRangeIndexes,
              }) => {
                mockAfterMenuClickFn({
                  type,
                  selectionRangeKeys,
                  selectionRangeIndexes,
                })
              },
              contextmenus: BODY_CONTEXTMENUS,
            },
          }
        },
      },
      // need attach to documnet
      { attachTo: document.body },
    )

    const firstTrTdEl = WRAPPER.findAll('.ve-table-body-tr')[2]
      .findAll('.ve-table-body-td')[2]

    firstTrTdEl.trigger('mousedown', {
      button: 3,
    })
    await later()
    firstTrTdEl.trigger('contextmenu')
    await later()

    expect(firstTrTdEl.classes()).toContain('ve-table-cell-selection')

    const contextmenuNodes = document.querySelectorAll(
      '.ve-contextmenu-node',
    )

    expect(contextmenuNodes.length).toBe(6)

    const contentmenuItemClickEvent = new MouseEvent('click', {
      view: window, // window
      bubbles: true,
      cancelable: true,
    })

    contextmenuNodes[2].dispatchEvent(contentmenuItemClickEvent)

    await later()

    expect(mockAfterMenuClickFn).toHaveBeenCalledWith({
      type: 'INSERT_ROW_ABOVE',
      selectionRangeKeys: {
        endColKey: 'date',
        endRowKey: 2,
        startColKey: 'date',
        startRowKey: 2,
      },
      selectionRangeIndexes: {
        endColIndex: 2,
        endRowIndex: 2,
        startColIndex: 2,
        startRowIndex: 2,
      },
    })
  })
})
