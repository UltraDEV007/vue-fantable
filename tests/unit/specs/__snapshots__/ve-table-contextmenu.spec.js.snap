// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FanTable contextmenu render 1`] = `
<div class="vue-table-root">
  <div class="fan-table fan-table-border-around" style="width: 100%;">
    <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col style="width: 50px;">
            <col style="width: 15%;">
            <col style="width: 15%;">
            <col style="width: 30%;">
            <col style="width: 40%;">
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header">
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-fixed-left fan-table-last-left-fixed-column" style="text-align: center; left: 0px; top: 0px;"></th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Name</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Date</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: right; top: 0px;">Number</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-last-column" style="text-align: left; top: 0px;">Address</th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-row-hover fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr row-key="0" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">1</td>
              <td rowspan="1" colspan="1" col-key="name" class="fan-table-body-td" style="text-align: left;">John</td>
              <td rowspan="1" colspan="1" col-key="date" class="fan-table-body-td" style="text-align: left;">1900-05-20</td>
              <td rowspan="1" colspan="1" col-key="number" class="fan-table-body-td" style="text-align: right;">32</td>
              <td rowspan="1" colspan="1" col-key="address" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shanghai</td>
            </tr>
            <tr row-key="1" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">2</td>
              <td rowspan="1" colspan="1" col-key="name" class="fan-table-body-td" style="text-align: left;">Dickerson</td>
              <td rowspan="1" colspan="1" col-key="date" class="fan-table-body-td" style="text-align: left;">1910-06-20</td>
              <td rowspan="1" colspan="1" col-key="number" class="fan-table-body-td" style="text-align: right;">676</td>
              <td rowspan="1" colspan="1" col-key="address" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Beijing</td>
            </tr>
            <tr row-key="2" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">3</td>
              <td rowspan="1" colspan="1" col-key="name" class="fan-table-body-td" style="text-align: left;">Larsen</td>
              <td rowspan="1" colspan="1" col-key="date" class="fan-table-body-td" style="text-align: left;">2000-07-20</td>
              <td rowspan="1" colspan="1" col-key="number" class="fan-table-body-td" style="text-align: right;">76</td>
              <td rowspan="1" colspan="1" col-key="address" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Chongqing</td>
            </tr>
            <tr row-key="3" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">4</td>
              <td rowspan="1" colspan="1" col-key="name" class="fan-table-body-td" style="text-align: left;">Geneva</td>
              <td rowspan="1" colspan="1" col-key="date" class="fan-table-body-td" style="text-align: left;">2010-08-20</td>
              <td rowspan="1" colspan="1" col-key="number" class="fan-table-body-td" style="text-align: right;">7797</td>
              <td rowspan="1" colspan="1" col-key="address" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Xiamen</td>
            </tr>
            <tr row-key="4" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">5</td>
              <td rowspan="1" colspan="1" col-key="name" class="fan-table-body-td" style="text-align: left;">Jami</td>
              <td rowspan="1" colspan="1" col-key="date" class="fan-table-body-td" style="text-align: left;">2020-09-20</td>
              <td rowspan="1" colspan="1" col-key="number" class="fan-table-body-td" style="text-align: right;">8978</td>
              <td rowspan="1" colspan="1" col-key="address" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shenzhen</td>
            </tr>
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
        </table>
        <div class="fan-table-selection-wrapper">
          <div class="fan-table-selection-fixed-left"></div>
          <div class="fan-table-selection-middle"></div>
          <div class="fan-table-selection-fixed-right"></div>
        </div>
      </div>
    </div>
    <div class="fan-table-edit-input-container" style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;"><textarea tabindex="-1" class="fan-table-edit-input"></textarea></div>
    <div class="ve-contextmenu" style="display: none;"></div>
  </div>
</div>
`;
